Firestore/core/src/firebase/firestore/remote/serializer.h

```diff
@@ -159,13 +164,24 @@ class Serializer {
   }
 
  private:
+  friend class local::LocalSerializer;
```

Comment:
Friends aren't a concept we have on the other platforms and there's really no reason not to just make the things the LocalSerializer uses public.

Reply:
It prevents users of the serializer from calling these (implementation detail) methods. The fact that the local serializer happens to use them at all is just a cheat that we can get away with due to the wire proto and disk proto not having sufficiently diverged yet.

These private methods also leak the implementation (nanopb). Mind you, that leak would also likely prevent any caller from accidentally using them, since they'd need to construct a nanopb::Reader/Writer in order to do so.

All-in-all; I'm still slightly in favour of `friend`, even if the other platforms are unable to take advantage of that.

The alternative is to address this via docs: `/** Do not use. Only public for use by the LocalSerializer. */`. I'm less thrilled, but still find this to be perfectly acceptable. (It's not like this is a public api or anything.)

(Another alternative is to refactor this common logic out to a NanopbSerializerHelper class and use that in both serializers. No friends, no needlessly public methods. But this diverges from the other platforms.)

Was the above convincing? If yes, I'll leave as is. If not, I'll switch the (relevant) private methods to public and add the doc string.

Reply:
What I disagree with is that need to be hidden from anybody. The serializer *itself* is an implementation detail. There's no reason to prevent anyone that needs to serialize a document to nanopb from doing so.

I also don't think we need to go to any lengths to hide the fact that the serializer uses nanopb. The comparable method in the Objective-C and Android-Java implementations returns an object bound to the specific proto library they use.

In short I think just making this public and being done with it is sufficient.